sammba/registration/struct.py

```diff
@@ -387,130 +424,87 @@ def anats_to_common(anat_filenames, write_dir, brain_volume,
-                              outputtype='NIFTI_GZ')
-    out_mask_tool = mask_tool(in_file=out_tcat.outputs.out_file, union=True,
-                              outputtype='NIFTI_GZ')
-    out_mask_tool = mask_tool(in_file=out_mask_tool.outputs.out_file,
-                              dilate_inputs='4',
-                              outputtype='NIFTI_GZ')
+    if nonlinear_weight_file is None:
+        out_mask_tool = mask_tool(in_file=out_tcat.outputs.out_file, union=True,
+                                  outputtype='NIFTI_GZ')
+        out_mask_tool = mask_tool(in_file=out_mask_tool.outputs.out_file,
+                                  dilate_inputs='4',
+                                  outputtype='NIFTI_GZ')
+        nonlinear_weight_file = out_mask_tool.outputs.out_file
 
     ###########################################################################
-    # The input source images are initially transformed prior to registration,
-    # to ensure that they are already quite well-aligned to the template.
-    # To save time, we only achieve one refinement level per step
+    # Description to fill
+    # 
+    # 
     if nonlinear_levels is None:
         nonlinear_levels = [1, 2, 3]
-
-    warped_files = []
-    warp_files = []
-    for affine_transform_file, centered_head_file in zip(
-            affine_transform_files, centered_head_files):
-        out_qwarp = qwarp(
-            in_file=centered_head_file,
-            base_file=out_tstat_allineated_head.outputs.out_file,
-            nmi=True,
-            noneg=True,
-            iwarp=True,
-            weight=out_mask_tool.outputs.out_file,
-            iniwarp=[affine_transform_file],
-            inilev=0,
-            maxlev=nonlinear_levels[0],
-            out_file=fname_presuffix(centered_head_file, suffix='_warped1'))
-        warp_files.append(out_qwarp.outputs.source_warp)
-        warped_files.append(out_qwarp.outputs.warped_source)
-
-    out_tcat = tcat(
-        in_files=warped_files,
-        out_file=os.path.join(write_dir, 'warped_1iter_heads.nii.gz'))
-    out_tstat_warp_head = tstat(in_file=out_tcat.outputs.out_file,
-                                outputtype='NIFTI_GZ')
-
-    ###########################################################################
-    # Then iterative registration from a given level to another is achieved.
-    # Note that any level below a patch size of 25 will not be done (see
-    # 3dQwarp help for further detail).
-    # The input transform is the former warp and needs to be concatenated to
-    # IDENT initially; I forget why, I think it is to avoid some weird bug.
-    if len(nonlinear_levels) > 1:
-        previous_warp_files = warp_files
+    if nonlinear_minimal_patches is None:
+       nonlinear_minimal_patches = []
+    levels_minpatches = nonlinear_levels + nonlinear_minimal_patches
+    out_tstat_warp_head = out_tstat_allineated_head # prevent later elif failure
```

Comment:
I think we shouldn't care about the interpreter complaint here. I think the case is too much complicated for it to see that `out_tstat_warp_head` will be created before called

Reply:
on my system it's more than a complaint: it's a stop error so there's no choice

Reply:
Ok. Let me just run the code with all your new changes on the rats data to try figure out why it is a stop error.

Reply:
@nadkarni-na I run your updated changes without this line and I have no error ! Could you give me the traceback of your error ?

Reply:
I cannot reproduce it anymore, I think it was a consequence of an overused terminal window